bitcoin-magazine/bitcoin-explained/explaining-bitcoin-addresses.md

```diff
@@ -0,0 +1,649 @@
+
+Yeah, which means it's unspendable.
+
+Aaron Van Wirdum: 00:27:17
+
+Right. Yeah, exactly.
+
+Sjors Provoost: 00:27:18
+
+But the good news is that there's another constraint for the original version of segwit, segwit version 0, which is that an address is either well 20 bytes or 32 bytes. Right. And that means that it's constrained right because if you add another queue to it then it's too long so you still know it's wrong
+
+Aaron Van Wirdum: 00:27:35
+
+yeah if you have a 20 byte address and you add one queue then it's 21 which is still invalid so you'd have to accidentally add 15 queues or how many were 12 well cues
+
+Sjors Provoost: 00:27:45
+
+yeah or something like that I don't know
+
+Aaron Van Wirdum: 00:27:47
+
+that's pretty unlikely to happen
+
+Sjors Provoost: 00:27:49
+
+yeah cuz I might be confusing bytes and characters, but exactly. That's very unlikely to happen for SegWit version 0. But now we would say, okay, we're going to have a new, you know, we're going to have future versions of SegWit, such as Taproot. Right. Which, you know, would be BC1P, because P is version 1. And I believe for Taproot, there's also a constraint in how long these addresses are supposed to be So it's still not an acute problem but in the future maybe we want to have addresses that are somewhat more arbitrary in length because maybe you want to add some weird conditions to it or you want to communicate other information not just the address maybe you want to put the amount inside the address.
+
+Aaron Van Wirdum: 00:28:30
+
+Right.
+
+Sjors Provoost: 00:28:31
+
+So this is why there's a new standard proposed BIP 350 which is called BEC32M and it's actually a very simple change. I think it adds to the all the math it adds one extra number to that math. And then it fixes that particular bug. And everybody's happy.
+
+Aaron Van Wirdum: 00:28:51
+
+So it fixes the bug that the queues don't matter anymore.
+
+Sjors Provoost: 00:28:56
+
+Yeah, you can't just add stuff to it without running into problems.
+
+Aaron Van Wirdum: 00:29:00
+
+But I guess this does mean that wallets that have by now upgraded to support these special SecWit addresses, BEC32 addresses, they now have to upgrade again.
+
+Sjors Provoost: 00:29:11
+
+That's right. So that's annoying, because it does mean that if your wallet wants to support sending to a taproot address, then it has to make a small change to the BEC32 implementation. And there's some example code on the BIP. It's not a big change because it just adds one number and if you look at the Bitcoin Core implementation it's a fairly simple change that does it. But it does mean that moving forward when you see a back 32 address you have to parse it then see if it's the version 0 or the version 1 and then do things slightly differently but even that is just a very small change right but it is annoying yeah it does mean that especially hardware wallets, you know, with firmware updates could take a while.
```

Comment:
```suggestion
That's right. So that's annoying, because it does mean that if your wallet wants to support sending to a taproot address, then it has to make a small change to the BECH32 implementation. And there's some example code on the BIP. It's not a big change because it just adds one number and if you look at the Bitcoin Core implementation it's a fairly simple change that does it. But it does mean that moving forward when you see a back 32 address you have to parse it then see if it's the version 0 or the version 1 and then do things slightly differently but even that is just a very small change right but it is annoying yeah it does mean that especially hardware wallets, you know, with firmware updates could take a while.
```